doc/man3/X509_check_purpose.pod

```diff
@@ -0,0 +1,74 @@
+=pod
+
+=head1 NAME
+
+X509_check_purpose - Check the purpose of a certificate
+
+=head1 SYNOPSIS
+
+ #include <openssl/x509v3.h>
+
+ int X509_check_purpose(X509 *x, int id, int ca)
+
+=head1 DESCRIPTION
+
+This function checks if certificate I<x> was created with the purpose
+represented by I<id>. If I<ca> is nonzero, then certificate I<x> is
+checked to determine if it's a possible CA with various levels of certainty
+possibly returned.
+
+Below are the potential ID's that can be checked:
+
+ # define X509_PURPOSE_SSL_CLIENT        1
+ # define X509_PURPOSE_SSL_SERVER        2
+ # define X509_PURPOSE_NS_SSL_SERVER     3
+ # define X509_PURPOSE_SMIME_SIGN        4
+ # define X509_PURPOSE_SMIME_ENCRYPT     5
+ # define X509_PURPOSE_CRL_SIGN          6
+ # define X509_PURPOSE_ANY               7
+ # define X509_PURPOSE_OCSP_HELPER       8
+ # define X509_PURPOSE_TIMESTAMP_SIGN    9
+
+=head1 RETURN VALUES
+
+For non-CA checks
+
+=over 4
+
+=item -1 an error condition has occured
+
+=item E<32>1 if the certificate was created to perform the purpose represented by I<id>
```

Comment:
excuse me, but what does this `<32>`  mean?
and who is `I <id>` looks like nonsense to me.

Reply:
`E<32>` means add a space character. This is (presumably) so that the "1" from -1 above and the 1 on this line, line up when rendered.

`id` is explained in the description above along with all of the potential values - so I'm not sure what you think is nonsense. It all makes sense to me.

Reply:
I did not know this convention, and i cannot see the end result.
I only read what is written here, and it does not make sense to me
because 32 is no space.

Reply:
Have you checked the generated man page, does it look right ?
Or are the metacharacters rendered correctly in html, 
what other formats do we have?

Reply:
Do other man pages also use this, or will it be the only one?

Reply:
and is 32 a space in EBCDIC ?

Reply:
> id is explained

but what is I<id> supposed to be?

Reply:
> I did not know this convention, and i cannot see the end result.

You can use `perldoc doc/man3/X509_check_purpose.pod` to see the output rendered as a man page, or `perldoc -ohtml doc/man3/X509_check_purpose.pod` to see it as HTML.

> I only read what is written here, and it does not make sense to me
because 32 is no space.

32 is the ASCII code for space.

> Have you checked the generated man page, does it look right ?

It does look right in the man page format. In the html format it literally inserts a space which is then ignored by the browser. Which is no worse than not having it - and since it renders well in `man` output, I think it is worth having.

> Do other man pages also use this, or will it be the only one?

Other man pages use the `E` tag - mostly for correct rendering of `<` and `>` characters. As far as I can see this is the first one to use it for a space character - but I don't see a problem with that.

> and is 32 a space in EBCDIC ?

No - but the number is explicitly defined as "The ASCII/Latin-1/Unicode character with that number.". See:
https://perldoc.perl.org/perlpod.html

So, as long as the perl system adheres to that, it should still work even on an EBCDIC system - although I don't have access to an EBCDIC system to actually check.

> but what is `I<id>` supposed to be?

I is the perldoc tag for italics. Historically we've used a mixture of B and I tags for denoting parameter names - but we seem to be increasingly standardising on using the I tag for such parameters.